client/txn.go

```diff
@@ -82,6 +82,17 @@ func newTxn(db DB, depth int) *Txn {
 	return txn
 }
 
+func newTxnFromProto(db DB, depth int, t proto.Transaction) *Txn {
+	txn := newTxn(db, depth)
+	txn.txn = t
+	return txn
+}
+
+// GetState returns the transaction protobuf.
+func (txn *Txn) GetState() *proto.Transaction {
```

Comment:
returning a proto.Transaction

I actually do like making txn exportable. Let me sending you another PR with that change.